CMakeLists.txt

```diff
@@ -0,0 +1,108 @@
+# Project information.
+cmake_minimum_required( VERSION 3.5.0 )
+project( AwsIotDeviceSdkEmbeddedC
+         VERSION 4.0.2
+         LANGUAGES C )
+
+# Allow the project to be organized into folders.
+set_property( GLOBAL PROPERTY USE_FOLDERS ON )
+
+# Use C90.
+set( CMAKE_C_STANDARD 90 )
+set( CMAKE_C_STANDARD_REQUIRED ON )
+
+# Do not allow in-source build.
+if( ${PROJECT_SOURCE_DIR} STREQUAL ${PROJECT_BINARY_DIR} )
+    message( FATAL_ERROR "In-source build is not allowed. Please build in a separate directory, such as ${PROJECT_SOURCE_DIR}/build." )
+endif()
+
+# Import global configurations.
+include("tools/cmake/filePaths.cmake")
+
+# Configure options to always show in CMake GUI.
+option( BUILD_TESTS
```

Comment:
I think we should avoid the need of this flag. We should be able to build both tests and demos, with each of those targets configured to include their appropriate config.h path, without needing this flag

Reply:
Do you mean removing this flag and just building BOTH tests and demos all the time? We had it setup in v4 beta_1 to only build demos when this flag is off. That is what I'm trying to do here as well. As an end user, I probably don't care about the tests anyway.

Reply:
I think if you build both demos and tests you will have to have CMock installed and ruby in order to create the mocks as part of the compilation of the demo, even if you are not interested in the tests at all. Is that a good idea?

I think a flag to disable building of the tests is probably a reasonable way to avoid this problem.

Reply:
Hmm, good point about CMock and Ruby dependencies